### DISCUSSION TITLE
Upload File Tab not avilable in Device Import to Upload CSV file

### DISCUSSION THREAD

**Discussion by mahmoudkhalil2** [netbox-community/netbox#11370] [Q&A]
Hello All, 

when i tried to import CSV file for Devices, the file upload tab not available and the available tabs is Racked Devices and Child Devices , my netbox version is v3.4.1 and also this tab is not available in the Netbox online demo.

How i can import CSV file for the devices.

<img width="884" alt="Upload File Tab Missed" src="https://user-images.githubusercontent.com/121564738/210390710-8641bd23-9c30-4827-b614-c15e06128cf7.PNG">

**Comment by candlerb** [comment]
You paste your CSV data directly into the "Data" field.

**Reply by mahmoudkhalil2** [reply]
The CSV data are tables and when copy and paste them, the data formats changed and not matched netbox format. How to paste the CSV data and match the data field format. And why upload csv is not available like other modules.

**Reply by candlerb** [reply]
What do you mean by "the data formats changed"?

The first line of the data should be the CSV headings, and lines 2 onwards are the data.  You choose your own order for the columns.

**Reply by candlerb** [reply]
(Sorry, previous comment deleted. I was actually on v3.3.10, because the v3.4.1 upgrade had not completed)

I can confirm the issue about the missing file upload:
* `/dcim/devices/import/` had these buttons on v3.3.10, but not on v3.4.1.
* `/ipam/ip-addresses/import/` had buttons on v3.3.10, which have moved to tabs on v3.4.1. (But the sample CSV heading row has gone).

![image](https://user-images.githubusercontent.com/44789/210435161-e2c67558-a63d-4be7-8c92-7728c754947d.png)

**Reply by mahmoudkhalil2** [reply]
Are  I need to downgrade my current version? or this issue will be solved soon?
How I can downgrade from my current version to your mentioned version.

**Reply by mahmoudkhalil2** [reply]
also the file upload tab is not available in the recent version 3.4.2 in the live demo portal, is their any plans to fix this issue and add the file upload tab in the device bulk import page.
@jeremystretch  @candlerb 

![image](https://user-images.githubusercontent.com/121564738/210533174-dcf0a471-5993-4f01-a9ad-dd4b42b21885.png)

**Comment by kkthxbye-code** [comment]
I think it's a regression from https://github.com/netbox-community/netbox/pull/10367

I think the file upload got moved to the base template tabs and the device bulk import is overwriting them here:

https://github.com/netbox-community/netbox/blob/1c72a80d9af29411b7ec89d3c1bb9909b41d1c72/netbox/templates/dcim/device_import.html#L4

@arthanson - Can you confirm?

**Reply by mahmoudkhalil2** [reply]
@kkthxbye-code  from which page in the Netbox GUI i can find the device CSV upload tab as you mentioned?

**Reply by candlerb** [reply]
> from which page in the Netbox GUI i can find the device CSV upload tab as you mentioned?

It should be where you're looking for it (`/dcim/devices/import/`), but due to a bug/oversight as [explained above](https://github.com/netbox-community/netbox/discussions/11370#discussioncomment-4585307), it's been accidentally removed from Netbox 3.4.  Until that bug is fixed, you'll need to copy your CSV data and paste it into the text area.

If that doesn't work for you, then that's a different problem, and you'll have to show the actual CSV data that you're pasting and the error that you're getting.  (Most likely you would get the same error if you were able to upload a CSV file anyway).

> How I can downgrade from my current version to your mentioned version.

You can't, because the database structure for 3.4.x is different to earlier versions.

If you upgraded from version 3.3.x or earlier, and you took a database backup before upgrading, then you can restore that database into a new installation of that version.

Otherwise, you can make a fresh installation of 3.3.10, but you'll lose any data you have already created.

**Reply by mahmoudkhalil2** [reply]
Thanks for your detailed reply, when we can expect this bug will be fix and the file upload tab to be available in device import page.

**Reply by kkthxbye-code** [reply]
> when we can expect this bug will be fix and the file upload tab to be available in device import page.

When someone fixes it. There's currently no issue for it, which would be the place to start.

**Comment by jeremystretch** [comment]
FYI this has been resolved under #11528.
